hal/src/thumbv7em/nvm.rs

```diff
@@ -305,64 +414,148 @@ impl Nvm {
     pub fn boot_protection(&mut self, protect: bool) -> Result<()> {
         // Check if requested state differs from current state
         if self.is_boot_protected() != protect {
-            // Wait until ready
-            while !self.is_ready() {}
-
             // Requires both command and key so the command is allowed to execute
-            if !protect {
-                // Issue Set boot protection disable (disable bootprotection)
-                self.command_sync(CMD_AW::SBPDIS);
-            } else {
+            if protect {
                 // Issue Clear boot protection disable (enable bootprotection)
-                self.command_sync(CMD_AW::CBPDIS);
+                self.command_sync(CMD_AW::CBPDIS)
+            } else {
+                // Issue Set boot protection disable (disable bootprotection)
+                self.command_sync(CMD_AW::SBPDIS)
             }
-
-            self.manage_error_states()
         } else {
-            Err(Error::NoChangeBootProtection)
+            Ok(())
         }
     }
 
-    /// Write to flash memory from a slice
+    /// Enable/disable region lock
     ///
-    /// # Safety
+    /// Flash memory is split into 32 regions. `u32` based mask allow to enable
+    /// (if bit is `1`) or disable (if bit is `0`) region lock in a given
```

Comment:
I think this condition should be inverted, bit 0 is region lock enabled. My bad.